runtime/autoload/health.vim

```diff
@@ -0,0 +1,433 @@
+function! s:trim(s) abort
+  return substitute(a:s, '^\_s*\|\_s*$', '', 'g')
+endfunction
+
+
+" Simple version comparison.
+function! s:version_cmp(a, b) abort
+  let a = split(a:a, '\.')
+  let b = split(a:b, '\.')
+
+  for i in range(len(a))
+    if a[i] > b[i]
+      return 1
+    elseif a[i] < b[i]
+      return -1
+    endif
+  endfor
+
+  return 0
+endfunction
+
+
+" Fetch the contents of a URL.
+function! s:download(url) abort
+  let content = ''
+  if executable('curl')
+    let content = system('curl -sL "'.a:url.'"')
+  endif
+
+  if empty(content) && executable('python')
+    let script = "
+          \try:\n
+          \    from urllib.request import urlopen\n
+          \except ImportError:\n
+          \    from urllib2 import urlopen\n
+          \\n
+          \try:\n
+          \    response = urlopen('".a:url."')\n
+          \    print(response.read().decode('utf8'))\n
+          \except Exception:\n
+          \    pass\n
+          \"
+    let content = system('python -c "'.script.'" 2>/dev/null')
+  endif
+
+  return content
+endfunction
+
+
+" Get the latest Neovim Python client version from PyPI.  The result is
+" cached.
```

Comment:
The caching is script-local.  It's gone after Neovim is restarted.  It was less about performance and more about being nice to pypi.python.org since I was running the command a lot.